src/server/hset_family.cc

```diff
@@ -1005,8 +1006,37 @@ void HSetFamily::HStrLen(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void StrVecEmplaceBack(StringVec& str_vec, const listpackEntry& lp) {
+  if (lp.sval) {
+    str_vec.emplace_back(reinterpret_cast<char*>(lp.sval), lp.slen);
+  } else {
+    str_vec.emplace_back(absl::StrCat(lp.lval));
+  }
+  return;
+}
+
 void HSetFamily::HRandField(CmdArgList args, ConnectionContext* cntx) {
+  if (args.size() > 3) {
+    DVLOG(1) << "Wrong number of command arguments: " << args.size();
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+
   string_view key = ArgS(args, 0);
+  int32_t count;
+  string_view with_values;
+
+  if (args.size() > 1) {
+    if (!SimpleAtoi(ArgS(args, 1), &count)) {
```

Comment:
fixed. 

regarding error on `count = 0`, this will make dragonfly behave differently from Redis. Redis still returns empty array in this case...

Reply:
> regarding error on count = 0, this will make dragonfly behave differently from Redis. Redis still returns empty array in this case...

Glad you looked, it's fine to keep it as is :)